Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +11,246 @@ public static class BalanceStats
         /// </summary>
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return "N/A.";
+            }
+
+            var maxBalance = 0m;
+            var name = new StringBuilder();
+
+            for (var i = 0; i < peopleAndBalances.Length; i++)
+            {
+                var peopleBalances = peopleAndBalances[i].Split(",");
+                var balance = GetLargestBalance(peopleBalances);
+                GetNamesWithHighestBalance(ref name, ref maxBalance, peopleBalances, balance);
```

Comment:
Avoid using ref, but I kind of understand the limitation of not being able to use objects, so it can slide.
However, you don't get the names here, you set them. So the name should represent that.

Reply:
About avoid using the ref with my GetNamesWithHighestBalance method, I am going to accept your offer to allow me to slide on this; I hope that you will teach us the better way to do in the future.